Source/WebCore/html/FormAssociatedElement.cpp

```diff
@@ -35,6 +35,18 @@ FormAssociatedElement::FormAssociatedElement(HTMLFormElement* form)
 {
 }
 
+HTMLFormElement* FormAssociatedElement::formForBindings() const
+{
+    HTMLFormElement* formElement = form();
+    if (!formElement)
+        return nullptr;
+
+    if (asHTMLElement().document().settings().shadowRootReferenceTargetEnabled() && &formElement->treeScope() != &asHTMLElement().treeScope())
+        return nullptr;
+
+    return formElement;
```

Comment:
Another way to write this could be:

```cpp
if (asHTMLElement().document().settings().shadowRootReferenceTargetEnabled()) {
    auto* form = this->form();
    if (!form || &form->treeScope() != &asHTMLElement().treeScope())
        return nullptr;
    return form;
}
return form();
```

I think there's a couple other places where this pattern could apply to.

Reply:
This question has been discussed in the past, and there are proponents for both sides. I personally lean more towards explicit type specification outside of complicated types like lambdas / iterators / etc, but the project has aligned on `auto`, so I go with that 🙂